compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -136,220 +136,224 @@ abstract class PrepJSInterop extends plugins.PluginComponent
     /** DefDefs in class templates that export methods to JavaScript */
     private val exporters = mutable.Map.empty[Symbol, mutable.ListBuffer[Tree]]
 
-    override def transform(tree: Tree): Tree = postTransform { tree match {
-      // Nothing is allowed in native JS classes and traits
-      case idef: ImplDef if enclosingOwner is OwnerKind.JSNativeClass =>
-        reporter.error(idef.pos, "Native JS traits and classes " +
-            "may not have inner traits, classes or objects")
-        super.transform(tree)
-
-      // Handle js.Anys
-      case idef: ImplDef if isJSAny(idef) =>
-        transformJSAny(idef)
-
-      /* In native JS objects, only js.Any stuff is allowed. However, synthetic
-       * companion objects need to be allowed as they get generated, when a
-       * native class inside a native JS object has default arguments in its
-       * constructor (see #1891).
-       */
-      case modDef: ModuleDef if (enclosingOwner is OwnerKind.JSNativeMod) &&
-          modDef.symbol.isSynthetic =>
-        super.transform(tree)
-
-      // In native JS objects, only js.Any stuff is allowed
-      case idef: ImplDef if enclosingOwner is OwnerKind.JSNativeMod =>
-        reporter.error(idef.pos, "Native JS objects cannot contain inner " +
-            "Scala traits, classes or objects (i.e., not extending js.Any)")
-        super.transform(tree)
-
-      // @ScalaJSDefined is only valid on a js.Any
-      case idef: ImplDef if idef.symbol.hasAnnotation(ScalaJSDefinedAnnotation) =>
-        reporter.error(idef.pos,
-            "@ScalaJSDefined is only allowed on classes extending js.Any")
-        super.transform(tree)
-
-      // Catch the definition of scala.Enumeration itself
-      case cldef: ClassDef if cldef.symbol == ScalaEnumClass =>
-        enterOwner(OwnerKind.EnumImpl) { super.transform(cldef) }
-
-      // Catch Scala Enumerations to transform calls to scala.Enumeration.Value
-      case idef: ImplDef if isScalaEnum(idef) =>
-        val kind =
-          if (idef.isInstanceOf[ModuleDef]) OwnerKind.EnumMod
-          else OwnerKind.EnumClass
-        enterOwner(kind) { super.transform(idef) }
-
-      // Catch (Scala) ClassDefs to forbid js.Anys
-      case cldef: ClassDef =>
-        val sym = cldef.symbol
-
-        if (sym.hasAnnotation(JSNativeAnnotation)) {
-          reporter.error(cldef.pos, "Traits and classes not extending js.Any " +
-              "may not have a @js.native annotation")
-        }
+    override def transform(tree: Tree): Tree = {
+      if (tree.symbol != null)
+        checkInternalAnnotations(tree.symbol)
+      postTransform { tree match {
```

Comment:
Consider using this opportunity to also change this to:
```scala
val preTransformedTree = tree match {
  ...
}
postTransform(preTransformedTree)
```
to get rid of the `} }` at the end of this match, which is not a style that we use.